solr/src/main/scala/org/apache/pekko/stream/connectors/solr/impl/SolrFlowStage.scala

```diff
@@ -114,8 +113,12 @@ private final class SolrFlowLogic[T, C](
 
       message.routingFieldValue.foreach { routingFieldValue =>
         val routingField = client match {
-          case csc: CloudSolrClient =>
-            Option(csc.getIdField)
+          case csc: CloudSolrClient => {
+            val docCollection = Option(csc.getZkStateReader.getClusterState.getCollectionOrNull(collection))
```

Comment:
minor: I think you can directly access the docCollection using `csc.getZkStateReader.getCollection`. The underlying call is checking for the `clusterState` and calls the `getCollectionOrNull`

Reply:
thanks - change made